src/core/currency/wallet/currency-wallet-api.ts

```diff
@@ -159,8 +159,14 @@ export function makeCurrencyWalletApi(
       denominatedAmount: string,
       currencyCode: string
     ): Promise<string> {
+      const plugin = input.props.state.plugins.currency[pluginId]
```

Comment:
We should not be calling `getBuiltinTokens` here, since this has already been done once at load-time. Instead, use `input.props.state.accounts[accountId].allTokens[pluginId]`. That gives you both custom tokens and builtin tokens merged into a single `EdgeTokenMap`. It automatically updates as the user edits custom tokens.

Then you can just change the signature to `getCurrencyMultiplier(currencyInfo: EdgeCurrencyInfo, allTokens: EdgeTokenMap, currencyCode: string): string`. We don't need to worry about metaTokens anymore, since it's redundant with `builtinTokens` / `allTokens` now.

Reply:
👍